pkg/jobrunaggregator/jobruntestcaseanalyzer/cmd.go

```diff
@@ -0,0 +1,290 @@
+	PayloadTag                  string
+	Timeout                     time.Duration
+	EstimatedJobStartTimeString string
+	Platform                    string
+	Infrastructure              string
+	Network                     string
+	MinimumSuccessfulInstall    int
+	PayloadInvocationID         string
+	JobGCSPrefixes              []jobGCSPrefix
+}
+
+func NewJobRunsTestCaseAnalyzerFlags() *JobRunsTestCaseAnalyzerFlags {
+	return &JobRunsTestCaseAnalyzerFlags{
+		DataCoordinates: jobrunaggregatorlib.NewBigQueryDataCoordinates(),
+		Authentication:  jobrunaggregatorlib.NewGoogleAuthenticationFlags(),
+
+		WorkingDir:                  "test-case-analyzer-working-dir",
+		EstimatedJobStartTimeString: time.Now().Format(kubeTimeSerializationLayout),
+		Timeout:                     3*time.Hour + 30*time.Minute,
+	}
+}
+
+const kubeTimeSerializationLayout = time.RFC3339
+
+func (f *JobRunsTestCaseAnalyzerFlags) BindFlags(fs *pflag.FlagSet) {
+	f.DataCoordinates.BindFlags(fs)
+	f.Authentication.BindFlags(fs)
+
+	fs.StringVar(&f.PayloadTag, "payload-tag", f.PayloadTag, "The payload tag to analyze test case status, like 4.9.0-0.ci-2021-07-19-185802")
+	fs.StringVar(&f.EstimatedJobStartTimeString, "job-start-time", f.EstimatedJobStartTimeString, fmt.Sprintf("Start time in RFC822Z: %s", kubeTimeSerializationLayout))
+	fs.StringVar(&f.Platform, "platform", f.Platform, "The platform used to narrow down a subset of the jobs to analyze, ex: aws|gcp|azure|vsphere")
+	fs.StringVar(&f.Infrastructure, "infrastructure", f.Infrastructure, "The infrastructure used to narrow down a subset of the jobs to analyze, ex: upi|ipi")
+	fs.StringVar(&f.Network, "network", f.Network, "The network used to narrow down a subset of the jobs to analyze, ex: sdn|ovn")
+	fs.IntVar(&f.MinimumSuccessfulInstall, "minimum-successful-install", 1, "minimum number of successful installations among jobs meeting criteria")
+	usage := fmt.Sprintf("mutually exclusive to --payload-tag.  Matches the .label[%s] on the prowjob, which is a UID", jobrunaggregatorlib.PayloadInvocationIDLabel)
+	fs.StringVar(&f.PayloadInvocationID, "payload-invocation-id", f.PayloadInvocationID, usage)
+
+	fs.StringVar(&f.WorkingDir, "working-dir", f.WorkingDir, "The directory to store caches, output, and the like.")
+	fs.DurationVar(&f.Timeout, "timeout", f.Timeout, "Time to wait for analyzing job to complete.")
+	fs.Var(&jobGCSPrefixSlice{&f.JobGCSPrefixes}, "explicit-gcs-prefixes", "a list of gcs prefixes for jobs created for payload. Only used by per PR payload promotion jobs. The format is comma-separated elements, each consisting of job name and gcs prefix separated by =, like openshift-machine-config-operator=3028-ci-4.11-e2e-aws-ovn-upgrade~logs/openshift-machine-config-operator-3028-ci-4.11-e2e-aws-ovn-upgrade")
+}
+
+func NewJobRunsTestCaseAnalyzerCommand() *cobra.Command {
+	f := NewJobRunsTestCaseAnalyzerFlags()
+
+	/* Example runs for release controller:
+	 ./job-run-aggregator analyze-required-install
```

Comment:
`analyzer-required-test` with a repeatable argument that allows specifying something like

`--required-test separated|||suite|||name.testCaseName,numPasses,<integer>` with separation logic (there's probably a CSV library?).

Reply:
I could probably also see a special string for "install" and "upgrade" since we are fairly likely to change those test names over time.